src/platform/Linux/bluez/BluezEndpoint.cpp

```diff
@@ -267,31 +268,41 @@ void BluezEndpoint::RegisterGattApplicationDone(GObject * aObject, GAsyncResult
                                                                             &error.GetReceiver());
 
     VerifyOrReturn(success == TRUE, {
-        ChipLogError(DeviceLayer, "FAIL: RegisterApplication : %s", error->message);
-        BLEManagerImpl::NotifyBLEPeripheralRegisterAppComplete(false);
+        ChipLogError(DeviceLayer, "FAIL: RegisterGattApplication: %s", error->message);
```

Comment:
> Could we re-use the logic somehow?

I can do it as a follow up. In next related PRs I've got a plan to extract BlueZ client manager to separate class/file (the responsibility of it will be to keep track of proxy objects and propagate object added/removed/changed signals to subscribers like advertiser or scanner). I can place a helper function in that separate file.

Reply:
> if/else here as well.

Done